src/coronawhy_vt/utils/local_processing.py

```diff
@@ -0,0 +1,375 @@
+        TODO seems like this should be pulled into the function as it depends
+        entirely on the block_selector argument
+    :param words: List of lemmatized words
+    :param replace_dict: Takes a dict of {token: lemmatized underscore token}
+        and replaces occurences of token in text with its value
+
+    :return: {'identifier feature': list of occurrences}
+    """
+    positions = []
+    found_words = []
+    section = []
+    block_id = []
+    block_text = []
+    paper_id = []
+
+    tokenized_words = []
+    for w in words:
+        tokenized_words.append(nlp(w.lower())[0].lemma_)
+    for path in paths:
+        files = os.listdir(path)
+        for p in files:
+
+            readfile = open(path / p, 'r')
+            paper = json.load(readfile)
+            blocks = block_selector(paper)
+
+            for b in range(0, len(blocks)):
+                text = blocks[b]['text'].lower()
+                if(replace_dict is not None):
+                    text = run_replace(text, replace_dict)
+                text = nlp(text)
+                for t in text:
+                    for w in tokenized_words:
+                        if(w == t.lemma_):
+                            section.append(sec_name)
+                            found_words.append(w)
+                            positions.append(t.idx)
+                            block_text.append(blocks[b]['text'])
+                            block_id.append(b)
+                            paper_id.append(p[:-5])
+    # TODO discuss pros/cons of this format vs a list of tuples
+    return {'sha': paper_id,
+            'blockid': block_id,
+            'word': found_words,
+            'sec': section,
+            'pos': positions,
+            'block': block_text}
+
+
+def pull_mentions_direct(
+        paths: list,
+        block_selector: Callable[[dict], list],
+        sec_name: str,
+        words: list,
+        replace_dict: dict = None) -> dict:
+    """Aggregate the positional features of the text corpus.
+
+    :param paths: List of pathlib.Path objects containing paths to the directories containing the corpus of text to
+        aggregate features from
+    :param block_selector: The function to retrieve the relevant block from text
+    :param sec_name: Corresponds with block_selector to select the relevant block of text. Options are 'title',
+        'abstract', 'body' TODO seems like this should be pulled into the function as it depends entirely on the
+        block_selector argument
+    :param words: List of lemmatized words
+    :param replace_dict: Takes a dict of {token: lemmatized underscore token} and replaces occurences of token in text
+        with its value
+
+    :return: {'identifier feature': list of occurrences}
+    """
+    # TODO reduce nesting?
+    positions = []
+    found_words = []
+    section = []
+    block_id = []
+    block_text = []
+    paper_id = []
+    words = [word.lower() for word in words]
```

Comment:
there's lots of overlap in this function versus the last one. Maybe there's the possibility to abstract?

Reply:
Considering moving some of these preprocessing functions into classes (sort of like how streamed tokenizers are done for spaCy and TensorFlow).

Reply:
In my experience if it doesn't get done when it's point out the first time, it will never get done. I'd appreciate if you take care of it now - it will make it easier for anyone else reviewing the code in the future too